### DISCUSSION TITLE
SearchField does not submit when hitting ENTER

### DISCUSSION THREAD

**Discussion by janvorwerk** [adobe/react-spectrum#5774] [Q&A] [answered]
Hello

Despite that the [doc states](https://react-spectrum.adobe.com/react-aria/SearchField.html#features) that `SearchField` (from `react-aria-components`) should trigger a form submission when pressing `Enter`, I do not manage to see this behavior.

Tested using Chrome, Firefox, Safari and Edge.

See this example:
https://codesandbox.io/p/sandbox/searchbuttonnotsubmitting-7cq2sy

Should I open an issue? or did I miss something 😭?

**Comment by LFDanLu** [comment, marked as answer]
Ah, that would because of this: https://github.com/adobe/react-spectrum/blob/b5974ad1ccf2abc984c5cc6664b3c2d03de36768/packages/%40react-aria/searchfield/src/useSearchField.ts#L60-L69. Its been a while since I touched that code, but I think we did it like that because we felt that typically users would want to trigger a search operation rather than submit a form when hitting enter.

**Reply by janvorwerk** [reply]
Thanks for your reply!  🙏 So looking at the code, I changed the sandbox referenced on the ticket to make it work, simply by forwarding the submit function to the search field:

```jsx
      <SearchField label="Search" name="search" onSubmit={onSubmit}>
```

Actually, this is [documented](https://react-spectrum.adobe.com/react-aria/SearchField.html#events): I need to learn and read your doc more throughly!! 🥹 It's quite dense! 👍
I read the features section too fast, reading `with support for the Enter and Escape keys to submit and clear the field` as "submit and clear the **form**"

**Reply by QzCurious** [reply]
Hi @LFDanLu, Remix user here. As Remix conventionally rely on how native form submission works, preventing default behavior is kind of troubled. Kindly asking you to check the purpose and explain more on meaning of "search operation". (I thought `<input type="search">` is mean to enter-trigger a form regardless of using Remix or not)
